File: uniffi/src/message/groups/trans_idx_sig_groups.rs
use cesrox::error::ParsideResult;
use cesrox::{Prefixer, Seqner};
use cesrox::Saider;
pub use cesrox::message::groups::{
    TransIdxSigGroups,
    TransIdxSigGroup,
    ControllerIdxSigs
};

pub fn trans_idx_sig_group_create(prefixer: Prefixer, seqner: Seqner, saider: Saider, isigers: ControllerIdxSigs) -> TransIdxSigGroup {
    TransIdxSigGroup::new(prefixer, seqner,saider ,isigers)
}

pub fn trans_idx_sig_groups_create(value: Vec<TransIdxSigGroup>) -> TransIdxSigGroups {
    TransIdxSigGroups::new(value)
}

pub fn trans_idx_sig_groups_qb64(trans_idx_sig_groups: &TransIdxSigGroups) -> ParsideResult<String> {
    trans_idx_sig_groups.qb64()
}

pub fn trans_idx_sig_groups_qb64b(trans_idx_sig_groups: &TransIdxSigGroups) -> ParsideResult<Vec<u8>> {
    trans_idx_sig_groups.qb64b()
}

pub fn trans_idx_sig_groups_qb2(trans_idx_sig_groups: &TransIdxSigGroups) -> ParsideResult<Vec<u8>> {
    trans_idx_sig_groups.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/TransIdxSigGroups.kt
package cesrox.groups

import cesrox.primitives.Prefixer
import cesrox.primitives.Seqner
import cesrox.primitives.Saider
import uniffi.cesrox.*
import uniffi.cesrox.TransIdxSigGroup as TransIdxSigGroupNative
import uniffi.cesrox.TransIdxSigGroups as TransIdxSigGroupsNative

class TransIdxSigGroup {
    val value: TransIdxSigGroupNative

    constructor(prefixer: Prefixer, seqner: Seqner, saider: Saider, isigers: ControllerIdxSigs) {
        value = transIdxSigGroupCreate(prefixer.value, seqner.value, saider.value, isigers.value)
    }

    constructor(nativeValue: TransIdxSigGroupNative) {
        value = nativeValue
    }

    val prefixer get() = Prefixer(value.prefixer)

    val seqner get() = Seqner(value.seqner)

    val saider get() = Saider(value.saider)

    val isigers get() = ControllerIdxSigs(value.isigers)
}

class TransIdxSigGroups {
    val value: TransIdxSigGroupsNative

    constructor(couples: List<TransIdxSigGroup>) {
        value = transIdxSigGroupsCreate(couples.map { it.value })
    }

    constructor(couples: TransIdxSigGroupsNative) {
        value = couples
    }

    fun get(index: Int): TransIdxSigGroup {
        return TransIdxSigGroup(value.value[index])
    }

    fun qb2(): List<UByte> {
        return transIdxSigGroupsQb2(value)
    }
    fun qb64b(): List<UByte> {
        return transIdxSigGroupsQb64b(value)
    }
    fun qb64(): String {
        return transIdxSigGroupsQb64(value)
    }
}